frontend/libs/studio-components/src/components/StudioProperty/StudioProperties.mdx

```diff
@@ -1,15 +1,44 @@
 import { Canvas, Meta } from '@storybook/blocks';
-import { Heading, Paragraph } from '@digdir/designsystemet-react';
 import * as StudioPropertyStories from './StudioProperty.stories';
+import { StudioHeading } from '../StudioHeading';
+import { StudioParagraph } from '../StudioParagraph';
+import { StudioList } from '../StudioList';
+import { PlusCircleIcon } from '@studio/icons';
 
 <Meta of={StudioPropertyStories} />
 
-<Heading level={1} size='small'>
+<StudioHeading level={1} size='small'>
   StudioProperty
-</Heading>
-<Paragraph>
-  StudioProperty is a composed component designed to presentation of related data elements, by
-  providing group, fieldset, and individual editable properties.
-</Paragraph>
+</StudioHeading>
+<StudioParagraph>
+  StudioProperty is a composed component designed to present configurable content. It may be used to
+  let the user view and edit complex data structures.
+</StudioParagraph>
+<StudioParagraph>
+  It provides the following elements:
+  <StudioList.Root>
+    <StudioList.Unordered>
+      <StudioList.Item>
+        `StudioProperty.Group`: Groups the list of `Fieldset` and `Button` components together,
+        providing proper spacing between the components.
+      </StudioList.Item>
+      <StudioList.Item>
+        `StudioProperty.Fieldset`: Editable state of a property. Its children should be form
+        elements to set up the given property. It may also contain a nested `StudioProperty.Group`
+        component.
+      </StudioList.Item>
+      <StudioList.Item>
+        `StudioProperty.Button`: A button element to represent a given property. When no value is
+        given, it defaults to only show the property name with a <PlusCircleIcon /> icon.
+      </StudioList.Item>
+    </StudioList.Unordered>
+  </StudioList.Root>
+</StudioParagraph>
+<StudioParagraph>
+  The `StudioProperty.Fieldset` and `StudioProperty.Button` are designed to switch between each
+  other, but this switching must be implemented by the consumer. The button should switch to a
+  fieldset when clicked, while the fieldset should contain a button in its `menubar` prop to switch
+  back to a button.
```

Comment:
Would it make sense adding an example here?

Reply:
If you mean a code example, I don't really want to add that here, since me may update the component in the future and forget to update the example. It's better to refer to the code itself. Unfortunately, the "view code" functionality in Storybook is still quite bad (if you have any idea on how to fix it, be my guest), but the consumer may always have a look at the real code of the `Preview` component in the `.stories.tsx` file.

Reply:
Sounds reasonable 🫡